examples/guides/mp-restful-webservice/README.adoc

```diff
@@ -295,284 +185,223 @@ We need to make sure JAX-RS searches for beans. This file accomplishes that.
-+
---
-[source,java]
-// _include::0-2:{greeting-message-class}[tag=getter]
-// _include::4-6:{greeting-message-class}[tag=setter]
-----
-    String getMessage() {
+
+    String getMessage() { // <4>
         return message.get();
     }
 
-    void setMessage(String message) {
+    void setMessage(String message) { // <5>
         this.message.set(message);
     }
+}
 ----
---
-
-=== Create a JAX-RS root resource for the application
+<1> Makes sure the system allocates one instance of `GreetingProvider` and uses
+that one wherever `GreetingProvider` is injected.
+<2> An `AtomicReference<String>` can handle potential concurrent attempts to update it,
+so one holds the greeting message. Your application might receive multiple concurrent HTTP requests that
+try to modify the message.
+<3> The constructor is annotated with `javax.inject.Inject` and
+accepts the initial message value as a `String` argument. The `ConfigProperty` annotation
+on that argument triggers automatic MicroProfile config processing to look up the
+`app.greeting` config value from (in our case) the default
+MicroProfile config source: `META-INF/microprofile-config.properties`.
+<4> Returns the greeting.
+<5> Sets the greeting.
+
+==== `GreetResource.java` - the JAX-RS root resource for the application
 This class defines the endpoints for the application.
 
-Add these imports:
-[source,java]
-// _include::0-10:{greet-resource-class}[tag=javaxImports]
-----
-import javax.enterprise.context.RequestScoped;
-import javax.inject.Inject;
-import javax.json.Json;
-import javax.json.JsonBuilderFactory;
-import javax.json.JsonObject;
-import javax.ws.rs.GET;
-import javax.ws.rs.PUT;
-import javax.ws.rs.Path;
-import javax.ws.rs.PathParam;
-import javax.ws.rs.Produces;
-import javax.ws.rs.core.MediaType;
-----
-
-. Create the `GreetResource.java` file.
-Mark it as request-scoped and declare the common path prefix that
+Note several points:
+
+. The resource is request-scoped and declares the common path prefix that
 all endpoints in the resource share.
 +
 --
 [source,java]
-// _include::0-2:{greet-resource-class}[tag=classDecl]
+// _include::0-2:{greet-resource-class}[tag=classDecl,indent=0]
 ----
 @Path("/greet")
 @RequestScoped
 public class GreetResource {
 }
 ----
 --
-. Create a private field for the application's `GreetingMessage`
+. JAX-RS injects the single instance of the `GreetingProvider` bean so the
+resource can access the greeting message.
 +
 --
 [source,java]
-// _include::0-1:{greet-resource-class}[tag=greetingMessageDecl]
+// _include::0-3:{greet-resource-class}[tag=ctor,indent=0]
 ----
-    @Inject
-    private GreetingMessage greeting;
+@Inject
+public GreetResource(GreetingProvider greetingConfig) {
+    this.greetingProvider = greetingConfig;
+}
 ----
-The system injects our application-scoped `GreetingMessage`
-instance.
 --
-. Write a private method to format the message your endpoints
+. A private method formats the messages the endpoints
```

Comment:
nit: ...formats the messages that the endpoints...

Reply:
done